In [11]:
import sys
import pandas as pd
import matplotlib

ModuleNotFoundError: No module named 'matplotlib'

In [2]:
sys.path.append('../scripts/')

import defaults as defs
# import dataCleaner as dc
# import dataVisualizer as dv
# cleaner = dc.dataCleaner('initial observation')
# visualizer = dv.dataVisualizer('initial observation')

In [3]:
df = pd.read_csv('../observations/structure.csv')

In [4]:
df[:7]

Assets  number_of_files  \
0  ed3071a667a11cc56e88ae0489bfe6aa               28   
1  4799763419d621cd41e7fb8abbcdd45d               17   
2  bf1f7af46eec0e92939a8b3ba51cbacd               46   
3  e68e20f592457b875ce29757ab855dfe               15   
4  6a8e741867d4f893afad015b77b52c39               16   
5  65299151bf4cbd1825d28e4a43d113ac                6   
6  42c5b4b367187c29ac8fdd78c70837d6               24   

                                           all_files  \
0  ['video.mp4', 'fs-click-endframe.png', 'rev-vi...   
1  ['thumbnail.jpg', 'landing_2.jpg', 'video.mp4'...   
2  ['f2-bg.png', 'f3-cta-rev-4.png', 'Line 1.png'...   
3  ['endtext.png', 'thumbnail.jpg', 'landing_2.jp...   
4  ['landing_2.jpg', 'video.mp4', 'cta.png', 'end...   
5  ['video.mp4', 'MPU-swipe-layer2.png', 'engagem...   
6  ['logo.png', 'arrowdown.png', 'star24.png', 's...   

                                              concat  
0  video.mp4 fs-click-endframe.png rev-video-cta-...  
1  thumbnail.jpg landing_2.jpg video.mp4 MPU-clic...  
2  f2-bg.png f3-cta-rev-4.png Line 1.png logo.png...  
3  endtext.png thumbnail.jpg landing_2.jpg endfra...  
4  landing_2.jpg video.mp4 cta.png end-1.jpg thum...  
5  video.mp4 MPU-swipe-layer2.png engagement_anim...  
6  logo.png arrowdown.png star24.png star33e.png ...

* Varying file numbers observed

In [8]:
print(df['number_of_files'].min(), df['number_of_files'].max(), df['number_of_files'].mean())

0 175 18.6939226519337


* Number of files range between 0 and 175, with the average being around 19

In [9]:
df['number_of_files'].value_counts()[:50].plot(kind='barh')

/tmp/ipykernel_201239/2481650620.py:1: FutureWarning: The behavior of `series[i:j]` with an integer-dtype index is deprecated. In a future version, this will be treated as *label-based* indexing, consistent with e.g. `series[i]` lookups. To retain the old behavior, use `series.iloc[i:j]`. To get the future behavior, use `series.loc[i:j]`.
  df['number_of_files'].value_counts()[:50].plot(kind='barh')


ImportError: matplotlib is required for plotting when the default backend "matplotlib" is selected.